docs/gen-docs/kyma_alpha_create_module.md

```diff
@@ -0,0 +1,53 @@
+---
+title: kyma alpha create module
+---
+
+Creates a module bundled as an OCI image from the contents of the given path
+
+## Synopsis
+
+Use this command to create a Kyma module.
+
+### Detailed description
+
+Kyma modules are individual components that can be deplyed into a Kyma runtime. Modules are built and distributed as OCI continer images. 
+This command provides the means to create such images out of a folder's contents.
+
+This command will create a component descriptor in the descriptor path (./mod as a default) and package all the contents on the provided content path as a single layer.
+Optionally, additional layers can be added with contents in other paths.
+
+Finally, if a registry is provided, the created module will be pushed.
+
+
+```bash
+kyma alpha create module MODULE_NAME MODULE_VERSION <CONTENT_PATH> [flags]
+```
+
+## Flags
+
+```bash
+      --clean                  Remove the mod-path folder and all its contents at the end.
+  -c, --credentials string     Basic authentication credentials for the given registry in the format user:password
+      --insecure               Use an insecure connection to access the registry.
+      --mod-path string        Specifies the path where the component descriptor and module packaging will be stored. If the path already has a descriptor use the overwrite flag to overwrite it (default "./mod")
+      --name-mapping string    Repository context name mapping. Possible values are: empty, urlPath or sha256-digest
```

Comment:
why do we need this customized? I think we should only allow for defualt urlPath mapping as sha256-digest would be created remotely on the oci registry only. What do you think?

Reply:
Good point, I added this because the reference implementation has it. But it can be left empty and will just do no mapping at all.

Reply:
I suggest we default to urlPath because of this logic in the current operators based on the ocm spec:
```go
func getOCIRef(repo *ocm.OCIRegistryRepository, descriptor *ocm.ComponentDescriptor, ref string) (*OCI, error) {
	layerRef := OCI{
		Repo: repo.BaseURL,
	}
	switch repo.ComponentNameMapping { //nolint:exhaustive
	case ocm.OCIRegistryURLPathMapping:
		repoSubpath := DefaultRepoSubdirectory
		if ext, found := descriptor.GetLabels().Get(
			fmt.Sprintf("%s%s", ocm.OCIRegistryURLPathMapping, "RepoSubpath")); found {
			repoSubpath = string(ext)
		}
		layerRef.Repo = fmt.Sprintf("%s/%s", repo.BaseURL, repoSubpath)
		layerRef.Name = descriptor.GetName()
		// if ref is not provided, we simply use the version of the descriptor, this will usually default
		// to a component version that is valid
		if ref == "" {
			layerRef.Ref = descriptor.GetVersion()
		} else {
			layerRef.Ref = ref
		}
	default:
		return nil, fmt.Errorf("error while parsing componentNameMapping %s: %w",
			repo.ComponentNameMapping, ErrComponentNameMappingNotSupported)
	}
	return &layerRef, nil
}
```

Reply:
Right, in that case I will just always use `urlPath` and drop the flag.

Reply:
All uses of mapping now use `urlPath` and the flag has been removed.